src/python/pants/option/global_options.py

```diff
@@ -1311,146 +1312,116 @@ def register_options(cls, register):
+        "--dynamic-ui-renderer",
+        default=DynamicUIRenderer.indicatif_spinner,
+        help="If `--dynamic-ui` is enabled, selects the renderer.",
+    )
+
+    tag = StrListOption(
+        "--tag",
+        help=(
+            "Include only targets with these tags (optional '+' prefix) or without these "
+            f"tags ('-' prefix). See {doc_url('advanced-target-selection')}."
+        ),
+    ).metavar("[+-]tag1,tag2,...")
+    exclude_target_regexp = StrListOption(
+        "--exclude-target-regexp",
+        help="Exclude targets that match these regexes. This does not impact file arguments.",
+    ).metavar("<regexp>")
+
+    files_not_found_behavior = EnumOption(
+        "--files-not-found-behavior",
+        default=FilesNotFoundBehavior.warn,
+        help="What to do when files and globs specified in BUILD files, such as in the "
+        "`sources` field, cannot be found. This happens when the files do not exist on "
+        "your machine or when they are ignored by the `--pants-ignore` option.",
+    ).advanced()
+
+    owners_not_found_behavior = EnumOption(
+        "--owners-not-found-behavior",
+        default=OwnersNotFoundBehavior.error,
+        help=(
+            "What to do when file arguments do not have any owning target. This happens when "
+            "there are no targets whose `sources` fields include the file argument."
+        ),
+    ).advanced()
+
+    build_patterns = StrListOption(
+        "--build-patterns",
+        default=["BUILD", "BUILD.*"],
+        help=(
+            "The naming scheme for BUILD files, i.e. where you define targets.\n\n"
+            "This only sets the naming scheme, not the directory paths to look for. To add "
+            "ignore patterns, use the option `[GLOBAL].build_ignore`.\n\n"
+            "You may also need to update the option `[tailor].build_file_name` so that it is "
+            "compatible with this option."
+        ),
+    ).advanced()
+
+    build_ignore = StrListOption(
+        "--build-ignore",
+        help=(
+            "Paths to ignore when identifying BUILD files.\n\n"
+            "This does not affect any other filesystem operations; use `--pants-ignore` for "
+            "that instead.\n\n"
+            "Patterns use the gitignore pattern syntax (https://git-scm.com/docs/gitignore)."
+        ),
+    ).advanced()
+    build_file_prelude_globs = StrListOption(
+        "--build-file-prelude-globs",
+        help=(
+            "Python files to evaluate and whose symbols should be exposed to all BUILD files. "
+            f"See {doc_url('macros')}."
+        ),
+    ).advanced()
+    subproject_roots = StrListOption(
+        "--subproject-roots",
+        help="Paths that correspond with build roots for any subproject that this "
+        "project depends on.",
+    ).advanced()
+
+    _loop_flag = "--loop"
+    loop = BoolOption(_loop_flag, help="Run goals continuously as file changes are detected.")
```

Comment:
This didn't have a default before, so we should inspect the code a little closer to make sure that's OK.

Reply:
the default for a bool option is `False`. But imo it's confusing to not require you to explicitly set the value.

Could you please add `default=False`? While we don't want to change any code, this doesn't change behavior and I think is worth the fix.

Reply:
It's fine if we do something like this fix in a followup though